app/gui/view/graph-editor/src/component/node/input/area.rs

```diff
@@ -890,23 +927,23 @@ ensogl::define_endpoints! {
     }
 
     Output {
-        pointer_style       (cursor::Style),
-        width               (f32),
-        expression          (ImString),
-        editing             (bool),
-        ports_visible       (bool),
-        body_hover          (bool),
-        on_port_press       (Crumbs),
-        on_port_hover       (Switch<Crumbs>),
-        on_port_type_change (Crumbs,Option<Type>),
-        on_port_code_update (Crumbs,ImString),
-        on_background_press (),
-        view_mode           (view::Mode),
+        pointer_style        (cursor::Style),
+        width                (f32),
+        edit_mode_expression (ImString),
```

Comment:
1. The name is not obvious. If we want to express the intention here, this should probably be `expression_in_edit_mode` instead.
2. We should have FRP output "expression" that changes when the expression of the node changes. I do not understand why we have output that works only in edit mode.

Reply:
I agree with the name change. As for the general output expression outside of edit mode, I don't believe that we want that, as it is more confusing than helpful.

The node is not able to directly modify the expression by itself. In order to preserve AST IDs (used by widgets as their identity), we want the modifications to be applied only within specific span tree fragments, as deep in the tree as possible. In order to apply those modifications, we need to regenerate the span-tree. That requires access to the execution context, which is provided by the graph controller. That way when you select a value through a widget, the node view can only request the controller to perform the required modification, then update the view accordingly.

The current `expression_in_edit_mode` output is really only used to drive the searcher, which is not operating on span trees at all. In fact, this expression update is mostly ignored by the graph controller, because that would cause it to be applied permanently, even if we cancel the edit. Instead it is passed straight to the searcher. The whole edit and submit/cancel flow is managed by the searcher controller.

Eventually the searcher will have to be adapted to work within the edit mode of each individual widget. Then this signal will be completley removed.

So, once the whole-node edit mode is gone (and it will be gone soon), the node will at most be able to emit its set expression's code representation at given moment. It is guaranteed to be the `expression.repr()` of whatever was last set using `set_expression`. We don't have any use for that, as you can already trivially get that information without node view involvement.

Reply:
I ended up removing this signal from node component already, and generating the expression for searcher in the graph editor. That should make it a little less confusing.